gtk4/src/subclass/font_chooser.rs

```diff
@@ -0,0 +1,196 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::FontChooser;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, GString, ObjectExt, Quark};
+use once_cell::sync::Lazy;
+use pango::{FontFace, FontFamily, FontMap};
+
+pub trait FontChooserImpl: ObjectImpl {
+    fn get_font_family(&self, font_chooser: &Self::Type) -> Option<FontFamily>;
+    fn get_font_face(&self, font_chooser: &Self::Type) -> Option<FontFace>;
+    fn get_font_size(&self, font_chooser: &Self::Type) -> i32;
+    // fn set_filter_func(&self, font_chooser: &Self::Type) -> Option<Box<dyn Fn(&FontFamily, &FontFace) -> bool + 'static>>;
```

Comment:
This is the wrong way around. The caller would pass a filter function in here.

You'd need to define a new type for that which wraps the function pointer, user_data and destroy notify, which can be called from safe Rust, etc.

I think we have such a case elsewhere in GTK already. The foreach function of GtkContainer or so

Reply:
Also this function is not optional. You have to add bindings for it.